### PR TITLE
Allow configuration to optionally trigger events in place of exceptions when hydrating.

### PR DESCRIPTION
**Pull request by trq**
This PR relies upon https://github.com/bravo3/orm/pull/12 being merged and I will at that stage update deps.

### CODE REVIEW

File: src/Bravo3/OrmBundle/DependencyInjection/OrmExtension.php [hunk 1]
```diff
@@ -37,6 +37,8 @@ public function load(array $configs, ContainerBuilder $container)
             $em->setFactory([self::ORM_FACT_CLASS, self::ORM_FACT_METHOD]);
         }
 
+        $container->setParameter('hydration_exceptions_as_events', $config['hydration_exceptions_as_events']);
```
Comment (jordonsc) [inline src/Bravo3/OrmBundle/DependencyInjection/OrmExtension.php]: This should be:
`$container->getDefinition('orm.em')->addArgument($config['hydration_exceptions_as_events']);`

And leave off the parameter in the services.yml file

File: src/Bravo3/OrmBundle/Resources/config/services.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ services:
         class: Bravo3\Orm\Drivers\Redis\RedisDriver
     orm.em:
         class: Bravo3\Orm\Services\EntityManager
-        arguments: [@orm.driver, %kernel.cache_dir%]
+        arguments: [@orm.driver, %kernel.cache_dir%, %hydration_exceptions_as_events%]
```
Comment (jordonsc) [inline src/Bravo3/OrmBundle/Resources/config/services.yml]: Not required

File: src/Bravo3/OrmBundle/Services/Factories/OrmFactory.php [hunk 3]
```diff
@@ -15,11 +15,12 @@ class OrmFactory
      * @param string          $cache_dir
      * @return EntityManager
      */
-    public static function createEntityManager(DriverInterface $driver, $cache_dir)
+    public static function createEntityManager(DriverInterface $driver, $cache_dir, $hydration_exceptions_as_events)
```
Comment (jordonsc) [inline src/Bravo3/OrmBundle/Services/Factories/OrmFactory.php]: Give `$hydration_exceptions_as_events` it a default value of `false` to prevent BC breaks

Comment (trq) [inline src/Bravo3/OrmBundle/Services/Factories/OrmFactory.php]: Doesn't this get a default value of false via this though? https://github.com/bravo3/orm-bundle/pull/2/files#diff-3b89a603bd5905789f4b5e2ede10db75R50. Still, it probably reads better if I also add false here.

**Pull request merged**